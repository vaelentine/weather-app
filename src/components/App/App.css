:root {
  --bg-image: url("/src/assets/background.jpg");
  --bg-color: rgba(0, 0, 0, 0.8);
  --text-color: #FCFBF9;
  --error-color: #EE2919;
}

body {
  margin: 0;
  background: var(--bg-image);
  /* box-shadow: inset 2000px 0 0 0 var(--bg-color); */
  font-family: 28px Verdana, Geneva, Tahoma, sans-serif;
  color: var(--text-color);
  background-size: 100% 100%;
  animation: Gradient 3s linear infinite alternate;
  -webkit-animation: Gradient 15s linear infinite alternate;
}

.Header {
  font-weight: bold;
  font-size: 42px;
}

.appContainer {
  height: 100vh;
  width: 100vw;
}

.App {
  
  text-align: center;

}

.missingKey {
padding: 10px;
text-align: center;
background: #c21010;
color: white;
font-size: 24px;
font-family: Arial, Helvetica, sans-serif;
}
