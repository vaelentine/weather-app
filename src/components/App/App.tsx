import React, { useState } from 'react'
import Search from '../Search/Search'
import WeatherView from '../WeatherView/WeatherView'
import Message from 'components/Message/Message'
import Footer from '../Footer/Footer'
import './App.css'
import getWeather from 'api/getWeather'

function App() {
  const [weatherResponse, setWeatherResponse]:any = useState(null) //add geoloc

  function handleChange(newValue:any) {
    setWeatherResponse(newValue)
  }
  const missingApiKey: boolean = process.env.REACT_APP_WEATHER_API_KEY === undefined
  return (
    <div className="appContainer">
      { missingApiKey && <div className="missingKey"> <p>Warning: It looks like your API key isn't configured. </p>
      <p>Review the readme file for instructions to set one up.</p></div>}
      <div className="App">
        <div className="Header">What's the weather in ...</div>
        <Search setWeatherData={handleChange}/>
        {weatherResponse && (
          weatherResponse.cod === 200
            ? (<WeatherView weatherData={weatherResponse}/>)
            : (
            <Message messageData={weatherResponse}/>
              )
        )}

        <Footer />
      </div>
    </div>
  )
}

export default App
