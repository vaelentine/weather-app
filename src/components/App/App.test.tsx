import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import App from './App';

describe('App', () => {
  it('should render a search container', () => {
    render(<App />);
    const search_container = document.querySelector('#SearchContainer')
    expect(search_container).not.toBeNull();
  })
  it('should render a weather container', () => {
    render(<App />);
    const weather_container = document.querySelector('#WeatherContainer')
    expect(weather_container).not.toBeNull();
  })
  it('should render a footer', () => {
    render(<App />);
    const footer_component = document.querySelector('#Footer')
    expect(footer_component).not.toBeNull();
  })
})
