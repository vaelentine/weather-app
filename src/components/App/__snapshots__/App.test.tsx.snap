// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render without crashing 1`] = `
<DocumentFragment>
  <div
    class="appContainer"
  >
    <div
      class="App"
    >
      <div
        class="Header"
      >
        What's the weather in ...
      </div>
      <div
        class="Search"
      >
        <input
          placeholder="Enter city"
          type="text"
          value=""
        />
        <button
          class="SearchButton"
        >
          Find out!
        </button>
      </div>
      <div
        class="weatherContainer"
      />
      <div
        class="footerContainer"
      >
        <footer
          class="Footer"
        >
          <span>
            <a
              href="https://www.linkedin.com/in/magdalene-geise-17555b7/"
            >
              Maggie Geise
            </a>
            , 2021.
            <a
              href="https://github.com/vaelentine/weather-app"
            >
               Code on Github
            </a>
          </span>
        </footer>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
